### PR TITLE
Cancel Event

### LINKED ISSUE #62: cancelEvent
- Endpoint URI: */event/{eventId}/deactivate*
- Http Method: *PUT*
- API Logic: 
    - check event id exists in DB
    - Only event is active and upcoming can be cancelled
    - Set event status to cancelled
- Authority
  - Current CC Head
- Progress
    - [x] development
    - [x] integration Test

### PR DESCRIPTION
**Pull request by junzai97**
Resolve #62

### CODE REVIEW

File: src/main/java/com/thirdcc/webapp/web/rest/EventResource.java [hunk 1]
```diff
@@ -130,4 +130,19 @@ public EventResource(EventService eventService) {
...
     }
+
+    /**
+     {@code POST  /event/{eventId}/deactivate} : cancel "id" event.
+     *
+     * @param eventId the id of the eventDTO to cancel.
+     * @return the {@link ResponseEntity} with status {@code 200 (OK)} and 400 ()}.
+     */
+    @PutMapping("/event/{eventId}/deactivate")
+    public ResponseEntity<EventDTO> cancelEvent(@PathVariable Long eventId ){
```
Comment (CarlosSia) [inline src/main/java/com/thirdcc/webapp/web/rest/EventResource.java:146]: need add @preAuthorize annotation to only let current CC head to cancel the event

File: src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java [hunk 2]
```diff
@@ -111,4 +116,24 @@ public Event findEventByIdAndNotCancelledStatus(Long id) {
         return eventRepository.findOneByIdAndStatusIn(id, eventStatuses)
             .orElseThrow(() -> new BadRequestException("Event not found, might be cancelled or not exist"));
     }
+
+    /**
+     * Cancel event by id.
+     *
+     * @param id the id of the entity.
+     */
+    public EventDTO cancelEventById(Long id) {
```
Comment (HowardYaw) [inline src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java:126]: Missing Override Annotation

**Review by HowardYaw — CHANGES_REQUESTED** [review]

**Comment by CarlosSia** [comment]
got 1 file has conflict

File: src/main/java/com/thirdcc/webapp/web/rest/EventResource.java (hunk 1 above)
Comment (HowardYaw) [reply src/main/java/com/thirdcc/webapp/web/rest/EventResource.java:146]: @junzai97  if fixed please resolved conversation

File: src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java (hunk 2 above)
Comment (junzai97) [reply src/main/java/com/thirdcc/webapp/service/impl/EventServiceImpl.java:126]: Resolved in fc1c175c9b7fbfb27dbf8eb8463c2f09acd3088b

**Review by HowardYaw — APPROVED** [review]

**Review by kachun333 — APPROVED** [review]

**Pull request merged**